src/main/java/pl/agh/cs/io/model/RuleRestriction.java

```diff
@@ -25,21 +27,33 @@ public RuleRestriction(WindowState state, long permittedNumSec, ExceededUsageAct
     }
 
     public void checkRestriction() {
-        if (rule != null && permittedNumSec > 0) {
-            double usedToday = ActivityTime.getActivityTimeFromList(rule.getTimes(), state);
+        if (rule != null && rule.getRestriction().get().permittedNumSec > 0) {
+            double usedToday = ActivityTime.getActivityTimeFromList(rule.getTimes(), rule.getRestriction().get().state);
 
-            if (usedToday > permittedNumSec) {
+            if (usedToday > rule.getRestriction().get().permittedNumSec) {
+                ;
                 long currentTime = Timestamp.valueOf(LocalDateTime.now()).getTime();
                 //notify or call method every numOfSecBetweenNotifications seconds after time is exceeded
-                if (currentTime > lastNotification + (1000 * numOfSecBetweenNotifications)) {
-                    lastNotification = currentTime;
-                    //TODO Notify user
-                    System.out.println("Time used up for " + rule.getExePath() + ", " + action);
+                if ( !isDisplayed &&
+                        currentTime > rule.getRestriction().get().lastNotification + (1000 * rule.getRestriction().get().numOfSecBetweenNotifications)) {
+
+                    rule.getRestriction().get().lastNotification = currentTime;
+                    this.isDisplayed = true;
+                    showAlert(rule);
+                    this.isDisplayed = false;
                 }
             }
         }
     }
 
+    public static void showAlert(Rule rule){
```

Comment:
Może lepiej byłoby to przenieść do nowej klasy i oddzielić wyświetlanie od logiki.